pkg/runutil/backoff.go

```diff
@@ -0,0 +1,276 @@
+package runutil
+
+import (
+	"context"
+	"math/rand"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+type BackoffType string
+
+const (
+	EXPONENTIAL BackoffType = "EXPONENTIAL"
+	SEQUENCE    BackoffType = "SEQUENCE"
+)
+
+type Config struct {
+	Type   BackoffType `yaml:"type"`
+	Config interface{} `yaml:"config"`
+}
+
+func NewBackoff(ctx context.Context, logger log.Logger, confContentYaml []byte) (Backoff, error) {
+	level.Info(logger).Log("msg", "loading backoff configuration")
```

Comment:
Any reason we are reimplementing this instead of using https://github.com/cenkalti/backoff or others? Have you investigated them and they do not work for you for some reason?